tools/src/bin/rust-indexer.rs

```diff
@@ -349,6 +349,40 @@ fn analyze_file(
     }
 }
 
+/// Uses heuristics to determine if the path is a Windows path,
+/// and if so, returns a Linux "equivalent". Not really equivalent,
+/// because there is no such thing, but equivalent enough for our
+/// purposes - that is, is_relative() returns a correct thing, and
+/// strip_prefix will work on relative paths.
+/// Note that this is a "linuxzed" path and not a "unixed" path
+/// this code should ever only really run inside the linux vagrant
+/// machine.
+fn linuxized_path(path: &PathBuf) -> PathBuf {
+    // Windows paths have backslashes and so on a Linux host will
+    // just have a single long component
+    if path.components().count() == 1 {
+        if let Some(pathstr) = path.to_str() {
+            if pathstr.find('\\').is_some() {
+                // Has a backslash, so probably a Windows path. Let's lean
+                // those slashes forward
+                let converted = pathstr.replace('\\', "/");
+                if converted.find(":/") == Some(1) {
+                    // Starts with a drive letter, so let's turn this into
+                    // an absolute path
+                    let abs = "/".to_string() + &converted;
+                    return PathBuf::from(abs);
+                } else {
```

Comment:
nit: No `else` after return? In this case the symmetry doesn't look that important, but if you think it is you can just do `return if ... { } else { };`

Reply:
Fixed, thanks